settings/src/script.js

```diff
@@ -60,31 +60,41 @@ function Main( { userId } ) {
 		hasPrimaryProvider,
 	} = user;
 	const [ globalNotice, setGlobalNotice ] = useState( '' );
+
+	const totpEnabled = record?.[ '2fa_available_providers' ].includes( 'Two_Factor_Totp' );
+	const backupCodesEnabled =
+		record?.[ '2fa_available_providers' ].includes( 'Two_Factor_Backup_Codes' );
```

Comment:
I also find it harder to review PRs that do unrelated things, or get too big. What do you think about splitting those clean-up commits into a different PR?

I like the idea of pulling all the `{provider}Enabled`, `hasPrimaryProvider`,  etc things into the `Context` object 👍🏻

Reply:
Thanks for pointing it out. I've split the DRYness commits into a separate PR, only related stuff is left in this one.

> I like the idea of pulling all the {provider}Enabled, hasPrimaryProvider, etc things into the Context object 👍🏻

IMHO, we might want to avoid using Context in this case as we're not dealing with `prop drilling` - passing data down through many levels of components, and also not passing down to _many_ components. Overusing Context sometimes makes the code hard to maintain.